action.yaml

```diff
@@ -138,65 +139,72 @@ runs:
-                  echo "     Attempting to get as much info as possible before exiting 1"
-                  set +x
-                  kubectl -n ${namespace} logs --tail=0 job/${jobName}  || true
-                  kubectl -n ${namespace} describe job/${jobName}  || true
-                  kubectl -n ${namespace} describe pod -l "job-name=${jobName}" || true
-                  exit 1
-              fi
-              echo "... waiting"
-              sleep 10
-              kubectl -n ${namespace} logs --tail=0 job/${jobName}  2> /dev/null && ready=true || true
+            if [[ "$(date +'%m/%d %H:%M')" > "${finaldate}" ]]; then
+              echo "     Err: Timeout waiting for pod to start"
+              yellAndExit1
+            fi
+            echo "... waiting"
+            sleep 10
+            # check if the Job is finally active, or maybe already done
+            jobPodPhases=$(kubectl -n ${namespace} get pod -l controller-uid=${jobUid} --sort-by=.metadata.creationTimestamp -o jsonpath='{.items[-1:].status.phase}')
+            # pods phase can be 'Pending' 'Running' 'Failed' 'Succeeded'
+            [[ ${jobPodPhases} != 'Pending' ]] && ready=true
           done
-          echo "     Job is running"
-        else
-          echo "     Job is already done (logs might not be available if it complete too long ago)"
-        fi
+          echo "     Job is either 'Running' 'Failed' 'Succeeded'"
 
-        # then try to get logs
-        echo "     Attempt to fetch logs"
-        echo "-----------------------------"
-        echo ""
-        kubectl -n ${namespace} logs -f job/${jobName} || true
-        echo ""
-        echo "-----------------------------"
+          echo "     Attempt to fetch logs"
+          echo "-----------------------------"
+          echo ""
+          kubectl -n ${namespace} logs --timestamps=true --follow pod/"$(kubectl -n ${namespace} get pod -l controller-uid=${jobUid} --sort-by=.metadata.creationTimestamp -o jsonpath='{.items[-1:].metadata.name}')" || true
+          echo ""
+          echo "-----------------------------"
+
+          echo "     Job replica is done, checking Job status"
+          # not elegant but the safest way to get the overall Job status as .failed and conditions start to get tricky
+          # to look into as long as more than backofflimit is not 0
+          # give 5s to Kubernetes to have time to update the job status
+          kubectl -n ${namespace} wait --for=condition=complete --timeout=60s job/${jobName} && complete=true || true
+          if [[ "${complete}" == "true" ]]; then
+            echo "     Job final state is 'complete', it ended with sucess"
+            exit 0
+          else
+            if [[ $(kubectl -n ${namespace} get job/${jobName} -o jsonpath='{.spec.backoffLimit}') != $(kubectl -n ${namespace} get job/${jobName} -o jsonpath='{.status.failed}') ]]; then
+              echo "     Job replica failed, loop to wait for the next replica"
+              else
```

Comment:
please decrease indentation here a bit, to better see a separation between then/else.